tests/common/tests.rs

```diff
@@ -916,3 +916,526 @@ async fn variable_auth_live_query() -> Result<(), Box<dyn std::error::Error>> {
+		// Verify response contains no error
+		assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+	}
+
+	// Test operations that SHOULD work with an expired session
+	// These operations will refresh the session expiration
+	let res = socket
+			.send_request(
+				"signup",
+				json!([{
+					"ns": NS,
+					"db": DB,
+					"sc": "scope",
+					"email": "another@email.com",
+					"pass": "pass",
+				}]),
+			)
+			.await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	// Verify response contains no error
+	assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+	// Wait two seconds for the session to expire
+	tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+	// The session must be expired now or we fail the test
+	let res = socket.send_request("query", json!(["SELECT VALUE working FROM test:1",])).await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	assert_eq!(res["error"], json!({"code": -32000, "message": "There was a problem with the database: The session has expired"}));
+	let res = socket
+		.send_request(
+			"signin",
+			json!(
+				[{
+					"ns": NS,
+					"db": DB,
+					"sc": "scope",
+					"email": "another@email.com",
+					"pass": "pass",
+				}]
+			),
+		)
+		.await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	// Verify response contains no error
+	assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+	// Wait two seconds for the session to expire
+	tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+	// The session must be expired now or we fail the test
+	let res = socket.send_request("query", json!(["SELECT VALUE working FROM test:1",])).await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	assert_eq!(res["error"], json!({"code": -32000, "message": "There was a problem with the database: The session has expired"}));
+
+	// This needs to be last operation as the session will no longer expire afterwards
+	let res = socket.send_request("authenticate", json!([root_token,])).await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	// Verify response contains no error
+	assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+
+	// Test passed
+	server.finish();
+	Ok(())
+}
+
+#[test(tokio::test)]
+async fn session_reauthentication() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
https://github.com/surrealdb/surrealdb/pull/3561/files#r1500809184

Reply:
Resolved in [213e158](https://github.com/surrealdb/surrealdb/pull/3561/commits/213e1585edbb87540b6871b830f7f999f80430b2).